public/app/features/datasources/state/actions.ts

```diff
@@ -271,6 +298,13 @@ export function updateDataSource(dataSource: DataSourceSettings) {
     ) => DataSourceSettings
   ) => {
     try {
+      if (config.featureToggles.grafanaAPIServerWithExperimentalAPIs) {
+        // TODO: Store response
+        const apiVersions = await getDatasourceAPIVersions();
```

Comment:
We should probably do this somewhere else (or even do this in the backend and return it as part of Grafana `bootConfig`).

Reply:
perhaps in this folder?  https://github.com/grafana/grafana/blob/main/public/app/features/apiserver/client.ts